prog/aws/nic.rb

```diff
@@ -67,13 +109,28 @@ class Prog::Aws::Nic < Prog::Base
       allocation_id = nic.nic_aws_resource&.eip_allocation_id
       client.release_address({allocation_id: allocation_id}) if allocation_id
     end
+    hop_delete_subnet
+  end
+
+  label def delete_subnet
+    ignore_invalid_nic do
+      client.delete_subnet({subnet_id: nic.nic_aws_resource.subnet_id})
+    end
     pop "nic destroyed"
   end
 
   def client
     @client ||= nic.private_subnet.location.location_credential.client
   end
 
+  def private_subnet
+    @private_subnet ||= nic.private_subnet
+  end
+
+  def subnet_az
```

Comment:
`subnet_az` and `nic_aws_resource.subnet_az` might return different values. It feels little confusing. We might rename it